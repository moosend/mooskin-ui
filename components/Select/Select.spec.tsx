import * as React from 'react';
import {
	Select,
	SelectContainer,
	SelectFilter,
	SelectIcon,
	SelectOption,
	SelectOptionList,
	SelectOverlay,
	SelectPlaceholder,
} from './Select';

import { mount, shallow } from 'enzyme';

describe('Select', () => {
	test('renders Select correctly', () => {
		const func = jest.fn();

		const tree = shallow(
			<Select onChange={func}>
				<SelectContainer>
					<SelectPlaceholder>Select an option</SelectPlaceholder>
					<SelectFilter />
					<SelectIcon />
				</SelectContainer>
				<SelectOptionList>
					<SelectOption value="1">Option 1</SelectOption>
					<SelectOption value="2">Option 2</SelectOption>
					<SelectOption value="3">Option 3</SelectOption>
					<SelectOption value="4">Option 4</SelectOption>
					<SelectOption value="5">Option 5</SelectOption>
				</SelectOptionList>
				<SelectOverlay />
			</Select>
		);
		expect(tree).toMatchSnapshot();
	});

	test('renders Option correctly and calls callback on click', () => {
		const func = jest.fn();

		const tree = mount(
			<SelectOption value="1" onClick={func}>
				Option1
			</SelectOption>
		);

		tree.find('StyledSelectOption').simulate('click');

		expect(func).toHaveBeenCalled();
	});

	test('renders Filter correctly and calls callback on change', () => {
		const func = jest.fn();

		const tree = mount(<SelectFilter onChange={func} />);

		tree
			.find('StyledSelectFilter')
			.first()
			.simulate('change', { target: { value: 'filter' } });

		expect(func).toHaveBeenCalled();
	});

	test('should be rendered as a multi select if the selected prop is an array', () => {
		const options = ['2', '4'];

		const func = jest.fn();

		const component = shallow(
			<Select onChange={func} showList selectedValue={options} dataLabel="select">
				<SelectContainer>
					<SelectPlaceholder>Select an option</SelectPlaceholder>
					<SelectFilter />
					<SelectIcon />
				</SelectContainer>
				<SelectOptionList>
					<SelectOption value="1">Option 1</SelectOption>
					<SelectOption value="2">Option 2</SelectOption>
					<SelectOption value="3">Option 3</SelectOption>
					<SelectOption value="4">Option 4</SelectOption>
					<SelectOption value="5">Option 5</SelectOption>
				</SelectOptionList>
				<SelectOverlay />
			</Select>
		);

		// component.find('StyledSelectOption').first().simulate('click');
		// expect(func).toHaveBeenCalledWith({dataLabel: 'select', value: ['2', '4', '1']});
		expect(component).toMatchSnapshot();
	});
});
