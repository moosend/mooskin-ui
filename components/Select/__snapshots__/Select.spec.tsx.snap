// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders Option correctly 1`] = `
<li
  onClick={[Function]}
>
  Option1
</li>
`;

exports[`Select renders Select correctly 1`] = `
<div
  className="select-component componentContainer  myClass"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <label
    className="label "
  >
    Select something
  </label>
  <div
    className="overlay"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <div
      className="selectContainer labelContainer undefined"
      style={
        Object {
          "zIndex": 0,
        }
      }
    >
      <input
        className="innerInput"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Type to filter"
        style={
          Object {
            "display": "none",
          }
        }
        type="text"
        value=""
      />
      <div
        className="label-container innerDiv"
        style={
          Object {
            "display": "block",
          }
        }
      >
        No options available
      </div>
      <div
        className="selectIcon"
      />
      <input
        readOnly={true}
        style={
          Object {
            "display": "none",
          }
        }
        value="option1"
      />
      <div
        className="options-container optionsContainer"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <ul />
      </div>
    </div>
    <i
      className="description undefined"
    >
      with description
    </i>
  </div>
</div>
`;
